src/hal/src/pso/graphics.rs

```diff
@@ -101,22 +69,63 @@ pub struct BakedStates {
     /// Static depth bounds.
     pub depth_bounds: Option<Range<f32>>,
 }
-
+#[derive(Debug)]
+/// Primitive Assembler describes how input data are fetched in the pipeline and formed into primitives before being sent into the fragment shader.
+pub enum PrimitiveAssembler<'a, B: Backend> {
```

Comment:
This structure looks beautiful now, thank you!

Reply:
Thanks! :)